Cargo.toml

```diff
@@ -26,22 +26,24 @@ include = [
 ]
 
 [dependencies]
-pyo3 = "0.18.1"
+pyo3 = "0.18.2"
 regex = "1.6.0"
 strum = { version = "0.24.1", features = ["derive"] }
 strum_macros = "0.24.3"
 serde_json = {version = "1.0.87", features = ["preserve_order"]}
 enum_dispatch = "0.3.8"
 serde = "1.0.147"
 indexmap = "1.9.1"
-mimalloc = { version = "0.1.30", default-features = false, optional = true }
 speedate = "0.7.0"
 ahash = "0.8.0"
 url = "2.3.1"
 # idna is already required by url, added here to be explicit
 idna = "0.3.0"
 base64 = "0.13.1"
 
+[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
```

Comment:
can we continue to use mimalloc with wasm32 now we've reverted to `0.1.30`?

Although if removing mimalloc reduces the binary size, I would guess the download + wasm loading time improvement would more than make up for the performance benefits of mimalloc.

Reply:
Rebased, let's see how much binay size it can save first.

Reply:
perfect.

Reply:
with mimalloc

```
-rw-r--r-- 1 runner docker 866544 Apr  4 09:44 pydantic_core-0.22.0-cp311-cp311-emscripten_3_1_32_wasm32.whl
```

without mimalloc

```
-rw-r--r-- 1 runner docker 846820 Apr  4 09:45 pydantic_core-0.22.0-cp311-cp311-emscripten_3_1_32_wasm32.whl
```

Doesn't save much binary size.

Reply:
~2.25% - humm, I think then if we can use mimalloc with wasm32, it would be better.